.github/workflows/main.yml

```diff
@@ -16,6 +18,11 @@ jobs:
     runs-on: windows-latest
 
     steps:
+    - uses: n1hility/cancel-previous-runs@v2
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+      if: "!startsWith(github.ref, 'refs/tags/') && github.ref != 'refs/heads/master' && github.ref != 'refs/heads/develop'"
```

Comment:
The purpose of this is to cancel all jobs already running for this PR branch once a new commit is pushed. the last line prevents canceling tests that run for develop/master. 

We have this exact same steps in the header checks tests that run on PRs: https://github.com/stan-dev/math/blob/develop/.github/workflows/header_checks.yml#L22